ocaml/xapi/xapi_host_helpers.ml

```diff
@@ -411,8 +411,7 @@ let consider_enabling_host_nolock ~__context =
       debug
```

Comment:
As revealed from the latest test result, this variable will be `true` in following case:
The current host is a pool supporter. The  toolstack on the pool coordinator restarted. The call flow (**on_master_restart** -> Xapi_host_helpers.**consider_enabling_host_request**) on the pool supporter would reach here. So it is considered as a system boot although the supporter had neither a host reboot nor a toolstack restart.

Reply:
Could it be fixed by adding an extra parameter in `Xapi_host_helpers.consider_enabling_host_request` to indicate that this caller is `on_master_restart `?

Reply:
I'm not sure why `consider_enabling_host` is called in `on_master_restart`. The former clearly assumes that it is called as part of the startup sequence. Need to do some archeology...